src/pages/Office/MoveDetails/MoveDetails.jsx

```diff
@@ -149,6 +150,15 @@ const MoveDetails = ({
       shipment.status === shipmentStatuses.CANCELED,
   );
 
+  const shipmentWithDestinationAddressChangeRequest = mtoShipments?.filter(
+    (shipment) => shipment?.deliveryAddressUpdate?.status === 'REQUESTED' && !shipment.deletedAt,
+  );
+  useEffect(() => {
+    const shipmentCount = shipmentWithDestinationAddressChangeRequest?.length || 0;
+    if (setShipmentsWithDeliveryAddressUpdateRequestedCount)
+      setShipmentsWithDeliveryAddressUpdateRequestedCount(shipmentCount);
+  }, [mtoShipments, shipmentWithDestinationAddressChangeRequest, setShipmentsWithDeliveryAddressUpdateRequestedCount]);
```

Comment:
There are several useEffect blocks in this PR with similar dependency arrays, so this won't be something I recommend delving into on this PR, but it's helpful to socialize on project using React:

It isn't always obvious until you get bit by weird rendering issues, but using arrays and objects in the dependency array effectively causes the useEffect block to run on _every_ render. React does not deep-compare JS objects, so two identical objects with different memory addresses are considered not-equal for dependency arrays.

I _do_ think it's possible to make this one work with non object dependencies, but it would require a little testing. Something like this:

```suggestion
  }, [shipmentWithDestinationAddressChangeRequest?.length, setShipmentsWithDeliveryAddressUpdateRequestedCount]);
```

`mtoShipments` isn't used inside the useEffect block. mtoShipments changing does (potentially) impact whether the number of `shipmentWithDestinationAddressChangeRequest`'s changes, so with the suggested change, it should still trigger when desired.

Does that make sense? Anything I'm overlooking?

Reply:
non-blocking, but very curious!